File: Facade/PHP/HotelKeeperFacade.php
<?php

require_once "VegRestaurant.php";
require_once "NonVegRestaurant.php";
require_once "VegNonBothRestaurant.php";

class HotelKeeperFacade {
    public function getVegMenu() {
        $vegRestaurant = new VegRestaurant;
        $vegRestaurant->getMenus()->describe();
    }

    public function getNonVegMenu() {
        $nonVegRestaurant = new NonVegRestaurant;
        $nonVegRestaurant->getMenus()->describe();
    }

    public function getVegNonMenu() {
        $vegNonBothRestaurant = new VegNonBothRestaurant;
        $vegNonBothRestaurant->getMenus()->describe();
    }
}

File: Facade/Ruby/HotelKeeperFacade.rb
require_relative "NonVegRestaurant"
require_relative "VegNonBothRestaurant"
require_relative "VegRestaurant"

class HotelKeeperFacade
    def getVegMenu
        vegRestaurant = VegRestaurant.new
        vegRestaurant.getMenus.describe
    end

    def getNonVegMenu
        nonVegRestaurant = NonVegRestaurant.new
        nonVegRestaurant.getMenus.describe
    end

    def getVegNonMenu
        vegNonBothRestaurant = VegNonBothRestaurant.new
        vegNonBothRestaurant.getMenus.describe
    end
end